### PR TITLE
feature(network-selection): left overs

### PR DESCRIPTION
**Pull request by banklesss**
## Description / Change(s) / Related issue(s)

- Header logic improvements
- Sell/Buy funnel updates

##### Ticket

YOMO-1812

### CODE REVIEW

**Review by michalszorad — APPROVED** [review]

**Comment by banklesss** [comment]
converted to draft to include a missing change

**Comment by banklesss** [comment]
ready for review again

File: apps/wallet-mobile/src/features/Exchange/useCases/CreateExchangeOrderScreen/CreateExchangeButton/CreateExchangeButton.tsx [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+  const styles = useStyles()
+  const {orderType} = useExchange()
+  const strings = useStrings()
+  const {
+    selected: {network},
+  } = useWalletManager()
+
+  const isPreprod = network === Chain.Network.Preprod
+  const isSancho = network === Chain.Network.Sancho
+
```
Comment (stackchain) [inline apps/wallet-mobile/src/features/Exchange/useCases/CreateExchangeOrderScreen/CreateExchangeButton/CreateExchangeButton.tsx]:
```suggestion
  const isPreprod = network === Chain.Network.Preprod
  const isSancho = network === Chain.Network.Sancho
  const isBuy = orderType === 'buy'

```

File: apps/wallet-mobile/src/features/Exchange/useCases/CreateExchangeOrderScreen/CreateExchangeButton/CreateExchangeButton.tsx [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+  const title =
+    isPreprod && orderType === 'buy'
+      ? strings.createOrderPreprodFaucetButtonText
+      : isSancho && orderType === 'buy'
+      ? strings.createOrderSanchonetFaucetButtonText
+      : strings.proceed
```
Comment (stackchain) [inline apps/wallet-mobile/src/features/Exchange/useCases/CreateExchangeOrderScreen/CreateExchangeButton/CreateExchangeButton.tsx]:
```suggestion
  const title =
    isBuy
      ? isPreprod
        ? strings.createOrderPreprodFaucetButtonText
        : isSancho
        ? strings.createOrderSanchonetFaucetButtonText
        : strings.proceed
      : strings.proceed
```

File: apps/wallet-mobile/src/features/Exchange/useCases/CreateExchangeOrderScreen/CreateExchangeButton/CreateExchangeButton.tsx [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+
+  const handleOnPress = () => {
+    if (isPreprod && orderType === 'buy') {
```
Comment (stackchain) [inline apps/wallet-mobile/src/features/Exchange/useCases/CreateExchangeOrderScreen/CreateExchangeButton/CreateExchangeButton.tsx]:
```suggestion
    if (isPreprod && isBuy) {
```

File: apps/wallet-mobile/src/features/Exchange/useCases/CreateExchangeOrderScreen/CreateExchangeButton/CreateExchangeButton.tsx [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+      handleOnPressOnPreprod()
+      return
+    }
+
+    if (isSancho && orderType === 'buy') {
```
Comment (stackchain) [inline apps/wallet-mobile/src/features/Exchange/useCases/CreateExchangeOrderScreen/CreateExchangeButton/CreateExchangeButton.tsx]:
```suggestion
    if (isSancho && isBuy) {
```

File: apps/wallet-mobile/src/features/Exchange/useCases/CreateExchangeOrderScreen/CreateExchangeButton/CreateExchangeButton.tsx [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+      handleOnPressOnSanchonet()
+      return
+    }
+
+    onPress()
+  }
+
+  const isButtonDisabled = !(isPreprod || (isSancho && orderType === 'buy')) && disabled
```
Comment (stackchain) [inline apps/wallet-mobile/src/features/Exchange/useCases/CreateExchangeOrderScreen/CreateExchangeButton/CreateExchangeButton.tsx]:
```suggestion
  const isButtonDisabled = !(isPreprod || (isSancho && isBuy)) && disabled
```

File: apps/wallet-mobile/src/features/Exchange/useCases/CreateExchangeOrderScreen/CreateExchangeButton/CreateExchangeButton.tsx [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+
+const useStyles = () => {
+  const styles = StyleSheet.create({
+    actions: {
+      paddingVertical: 16,
+      paddingHorizontal: 16,
+    },
+  })
+  return styles
+}
```
Comment (stackchain) [inline apps/wallet-mobile/src/features/Exchange/useCases/CreateExchangeOrderScreen/CreateExchangeButton/CreateExchangeButton.tsx]:
```suggestion
const useStyles = () => {
  const {atoms} = useTheme()
  const styles = StyleSheet.create({
    actions: {
       ...atoms.p_lg,
    },
  })
  return styles
}
```

**Review by stackchain — CHANGES_REQUESTED** [review]
minor suggestion (it will require to import useTheme)

**Comment by banklesss** [comment]
done @stackchain

**Pull request merged**